Convert `embed hidden` into a proper boolean attribute

Ahmad-S792 opened the pull request:
#### 54238fd39a0d767f3fc4c69e128b1324d6007197
<pre>
Convert `embed hidden` into a proper boolean attribute

<a href="https://bugs.webkit.org/show_bug.cgi?id=259550">https://bugs.webkit.org/show_bug.cgi?id=259550</a>

Reviewed by Tim Nguyen.

This patch aligns WebKit with Blink / Chromium.

Inspired by: <a href="https://chromium-review.googlesource.com/c/chromium/src/+/3565492">https://chromium-review.googlesource.com/c/chromium/src/+/3565492</a>

This patch makes the &apos;hidden&apos; attribute behaves as a normal
boolean attribute - presence of the attribute is interpreted as
a &quot;true&quot; value.

This appears to be per-spec:

- <a href="https://html.spec.whatwg.org/#the-embed-element">https://html.spec.whatwg.org/#the-embed-element</a>
- <a href="https://html.spec.whatwg.org/#the-hidden-attribute">https://html.spec.whatwg.org/#the-hidden-attribute</a>
- <a href="https://html.spec.whatwg.org/#boolean-attribute">https://html.spec.whatwg.org/#boolean-attribute</a>

* Source/WebCore/html/HTMLEmbedElement.cpp:
(HTMLEmbedElement::collectPresentationalHintsForAttribute): Remove &apos;attribute&apos; cases and add assert for &apos;value&apos;
* LayoutTests/TestExpectations: Remove failing test case

Canonical link: <a href="https://commits.webkit.org/266399@main">https://commits.webkit.org/266399@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/86acd54f8752791ca9670aeb2663bd93257ad24f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13727 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14041 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14374 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15463 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13035 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13808 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16549 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14122 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15715 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13893 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14514 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11622 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16162 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11803 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12380 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/19414 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12878 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12545 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15757 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13072 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10949 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12340 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3338 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16673 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12913 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/html/HTMLEmbedElement.cpp:
```diff
@@ -82,10 +82,8 @@ RenderWidget* HTMLEmbedElement::renderWidgetLoadingPlugin() const
-            addPropertyToPresentationalHintStyle(style, CSSPropertyWidth, 0, CSSUnitType::CSS_PX);
-            addPropertyToPresentationalHintStyle(style, CSSPropertyHeight, 0, CSSUnitType::CSS_PX);
-        }
+        addPropertyToPresentationalHintStyle(style, CSSPropertyWidth, 0, CSSUnitType::CSS_PX);
+        addPropertyToPresentationalHintStyle(style, CSSPropertyHeight, 0, CSSUnitType::CSS_PX);
```
```suggestion
        ASSERT(!value.isNull());
        addPropertyToPresentationalHintStyle(style, CSSPropertyWidth, 0, CSSUnitType::CSS_PX);
        addPropertyToPresentationalHintStyle(style, CSSPropertyHeight, 0, CSSUnitType::CSS_PX);
```
I wonder if we should assert this, it would be wrong to apply this if the value is null (e.g. no hidden attribute present).

@rniwa Wdyt?

Ahmad-S792 replied:
Good point IMO!. Will wait for @rniwa input but happy to do change.

rniwa replied:
Yeah, asserting that makes sene to me.

Ahmad-S792 replied:
Done.

nt1m reviewed (approved):
r=me with assert added

Ahmad-S792 committed:
Convert `embed hidden` into a proper boolean attribute

https://bugs.webkit.org/show_bug.cgi?id=259550

Reviewed by Tim Nguyen.

This patch aligns WebKit with Blink / Chromium.

Inspired by: https://chromium-review.googlesource.com/c/chromium/src/+/3565492

This patch makes the 'hidden' attribute behaves as a normal
boolean attribute - presence of the attribute is interpreted as
a "true" value.

This appears to be per-spec:

- https://html.spec.whatwg.org/#the-embed-element
- https://html.spec.whatwg.org/#the-hidden-attribute
- https://html.spec.whatwg.org/#boolean-attribute

* Source/WebCore/html/HTMLEmbedElement.cpp:
(HTMLEmbedElement::collectPresentationalHintsForAttribute): Remove 'attribute' cases and add assert for 'value'
* LayoutTests/TestExpectations: Remove failing test case

Canonical link: https://commits.webkit.org/266399@main
